ddl/cluster_test.go

```diff
@@ -99,3 +107,72 @@ func TestGetFlashbackKeyRanges(t *testing.T) {
 	require.NoError(t, err)
 	require.Len(t, kvRanges, 1)
 }
+
+func TestFlashbackCloseAndResetPDSchedule(t *testing.T) {
+	store, dom := testkit.CreateMockStoreAndDomain(t)
+	originHook := dom.DDL().GetHook()
+	tk := testkit.NewTestKit(t, store)
+
+	oldValue := map[string]interface{}{
+		"hot-region-schedule-limit": 1,
+	}
+	infosync.SetPDScheduleConfig(context.Background(), oldValue)
+
+	timeBeforeDrop, _, safePointSQL, resetGC := MockGC(tk)
+	defer resetGC()
+	tk.MustExec(fmt.Sprintf(safePointSQL, timeBeforeDrop))
+
+	hook := &ddl.TestDDLCallback{Do: dom}
+	hook.OnJobRunBeforeExported = func(job *model.Job) {
+		assert.Equal(t, model.ActionFlashbackCluster, job.Type)
+		if job.SchemaState == model.StateWriteReorganization {
+			closeValue, err := infosync.GetPDScheduleConfig(context.Background())
+			assert.NoError(t, err)
+			assert.Equal(t, closeValue["hot-region-schedule-limit"], 0)
+			// cancel flashback job
+			job.State = model.JobStateCancelled
+			job.Error = dbterror.ErrCancelledDDLJob
+		}
+	}
+	dom.DDL().SetHook(hook)
+
+	ts, err := tk.Session().GetStore().GetOracle().GetTimestamp(context.Background(), &oracle.Option{})
+	require.NoError(t, err)
+
+	tk.MustGetErrCode(fmt.Sprintf("flashback cluster as of timestamp '%s'", oracle.GetTimeFromTS(ts)), errno.ErrCancelledDDLJob)
+	dom.DDL().SetHook(originHook)
+
+	finishValue, err := infosync.GetPDScheduleConfig(context.Background())
+	require.NoError(t, err)
+	require.EqualValues(t, finishValue["hot-region-schedule-limit"], 1)
+}
+
+func TestCancelFlashbackCluster(t *testing.T) {
+	store, dom := testkit.CreateMockStoreAndDomain(t)
+	originHook := dom.DDL().GetHook()
+	tk := testkit.NewTestKit(t, store)
+	ts, err := tk.Session().GetStore().GetOracle().GetTimestamp(context.Background(), &oracle.Option{})
+	require.NoError(t, err)
+
+	timeBeforeDrop, _, safePointSQL, resetGC := MockGC(tk)
+	defer resetGC()
+	tk.MustExec(fmt.Sprintf(safePointSQL, timeBeforeDrop))
+
+	// Try canceled on StateNone, cancel success
+	hook := newCancelJobHook(t, store, dom, func(job *model.Job) bool {
+		return job.SchemaState == model.StateNone
+	})
+	dom.DDL().SetHook(hook)
+	tk.MustGetErrCode(fmt.Sprintf("flashback cluster as of timestamp '%s'", oracle.GetTimeFromTS(ts)), errno.ErrCancelledDDLJob)
+	hook.MustCancelDone(t)
+
+	// Try canceled on StateWriteReorganization, cancel failed
+	hook = newCancelJobHook(t, store, dom, func(job *model.Job) bool {
+		return job.SchemaState == model.StateWriteReorganization
+	})
+	dom.DDL().SetHook(hook)
+	tk.MustExec(fmt.Sprintf("flashback cluster as of timestamp '%s'", oracle.GetTimeFromTS(ts)))
```

Comment:
Should be `tk.MustGetErrCode(`

Reply:
It will run success now, because we can't cancel flashback job in `writeReorg` state.
But if we support sendRPC, it will failed. I will modify it later.

Reply:
Then there is a prbolem, plz check the unit test failure.

Reply:
Seems not related with flashback

```
fastbuild/bin/table/tables/tables_test_/tables_test.runfiles/__main__/table/tables/cache_test.go:234
[2022-09-05T02:16:49.774Z]         	Error:      	Not equal: 
[2022-09-05T02:16:49.774Z]         	            	expected: "[1 101 1001]\n[2 222 222]\n[3 113 1003]\n"
[2022-09-05T02:16:49.774Z]         	            	actual  : "[1 101 1001]\n[2 222 3333]\n[3 113 1003]\n"
[2022-09-05T02:16:49.774Z]         	            	
[2022-09-05T02:16:49.774Z]         	            	Diff:
[2022-09-05T02:16:49.774Z]         	            	--- Expected
[2022-09-05T02:16:49.774Z]         	            	+++ Actual
[2022-09-05T02:16:49.774Z]         	            	@@ -1,3 +1,3 @@
[2022-09-05T02:16:49.774Z]         	            	 [1 101 1001]
[2022-09-05T02:16:49.774Z]         	            	-[2 222 222]
[2022-09-05T02:16:49.774Z]         	            	+[2 222 3333]
[2022-09-05T02:16:49.774Z]         	            	 [3 113 1003]
[2022-09-05T02:16:49.774Z]         	Test:       	TestCacheTableBasicReadAndWrite
[2022-09-05T02:16:49.774Z]         	Messages:   	sql:select * from write_tmp1, args:[]
```

Reply:
Done, PTAL